### ISSUE TITLE
GCMC mini-batch training is not runnable on the latest version `0.8.0post1`

### ISSUE THREAD

**Issue by tonyjie** [dmlc/dgl#3819]
## 🐛 Bug

I'm running [GCMC example](https://github.com/dmlc/dgl/tree/master/examples/pytorch/gcmc) here and try if it's runnable on the latest version `0.8.0post1`. It turns out that [`train.py`](https://github.com/dmlc/dgl/blob/master/examples/pytorch/gcmc/train.py) is runnable, but [`train_sampling.py`](https://github.com/dmlc/dgl/blob/master/examples/pytorch/gcmc/train_sampling.py) (the mini-batch training script) is not runnbale on the latest version, while it could be runned on the last version `0.7.2`. 

Some arguments of the DataLoader have changed in the latest version, and still not updated in this training script. 

## To Reproduce

Just follow the README.md under the GCMC example. Try to run mini-batch training script. 

1. Using the latest DGL version > `0.8.0`
```
python3 train_sampling.py --data_name=ml-100k --use_one_hot_fea --gcn_agg_accum=stack --gpu 0
```

2. The error message would be a simple FAIL because of `unexpected keyword argument`. 
```
Traceback (most recent call last):
  File "train_sampling.py", line 378, in <module>
    run(0, n_gpus, args, devices, dataset)
  File "train_sampling.py", line 202, in run
    sampler = dgl.dataloading.MultiLayerNeighborSampler([None], return_eids=True)
TypeError: __init__() got an unexpected keyword argument 'return_eids'
```

3. If I remove this argument `return_eids=True` manually, there would be some other argument errors still for `EdgeDataLoader`. 
```
Traceback (most recent call last):
  File "train_sampling.py", line 378, in <module>
    run(0, n_gpus, args, devices, dataset)
  File "train_sampling.py", line 216, in run
    valid_dataloader = dgl.dataloading.EdgeDataLoader(
......
TypeError: __init__() got an unexpected keyword argument 'g_sampling'
```

## Expected behavior

It is runnable and normal in DGL `0.7.2`. The script should be updated to meet the latest functions in DGL `0.8.0`. 

## Environment

 - DGL Version (e.g., 1.0): 0.8.0post1
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): Pytorch 1.10.2
 - OS (e.g., Linux): Linux
 - How you installed DGL (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.8
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

**Comment by BarclayII** [comment]
This example is very outdated, I think we need to rewrite it with our new features.

**Comment by TristonC** [comment]
It should have been fixed in the 0.9.0 release.
